feat: Add function to check if AKS cluster has at least 3 nodepools

noyasarko opened the pull request:
The code changes introduce a new function, `IsPoolMin3Nodes()`, which checks if an AKS cluster has at least 3 nodepools for high availability. This function is added to the `AKSClusterCheck` class and enhances the AKS cluster assessment functionality.

Closes #24  
(teams message, I drafted cz am not sure)

Commit:
feat: Add function to check if AKS cluster has at least 3 nodepools

The code changes introduce a new function, `IsPoolMin3Nodes()`, which checks if an AKS cluster has at least 3 nodepools for high availability. This function is added to the `AKSClusterCheck` class and enhances the AKS cluster assessment functionality.

Nicwalle commented on AKS/AKSClusterCheck.psm1:
```diff
@@ -41,6 +41,17 @@ class AKSClusterCheck: ResourceCheck {
+        return $true
+    }
+    
+    #create new function that goes through nodepools and checks the minimum if it has atleast 3 nodes. Range od min-max if min is 1 thats not good. If 1 nodepool has min<3 the entire test fails
```
As a good practice, comments that are not part of the documentation should be removed from the code.

Nicwalle commented on AKS/AKSClusterCheck.psm1:
```diff
@@ -41,6 +41,17 @@ class AKSClusterCheck: ResourceCheck {
         return $totalNodeCount
     }
 
+    [bool] IsPoolMin3Nodes() {
```
For code clarity, please follow the same convention when it comes to case and naming. I would propose something like `hasMin3NodesPerPool`

Nicwalle commented on AKS/aksRules.json:
```diff
@@ -1,4 +1,9 @@
 {
+  "AKS_AmountOfNodesinPools": {
+    "expected": true,
+    "explanation": "The cluster should have at least 3 nodepools for high availability.",
+    "function": "IsPoolMin3Nodes"
+  },
```
For clarity, as it is resiliency related, can you please rename the check to it follows the convention SERVICE_DOMAIN_CHECK. Ex: `AKS_Resiliency_MinNodesPerPool`.
Would also be great if you can put this check with the other resiliency-related tests.

Commit:
feat: Refactor AKS cluster check function names

The code changes rename the functions in the `AKSClusterCheck` class to improve clarity and consistency. The function `IsPoolMin3Nodes()` is renamed to `hasMin3NodesPerPool()`, aligning with the naming convention used for other functions in the class. This refactoring enhances the readability and maintainability of the code.

Commit:
feat: Refactor AKS cluster check function names

Nicwalle committed:
feat: Improve AKS cluster check function names and explanations
